processing/src/main/java/org/apache/druid/segment/UnnestStorageAdapter.java

```diff
@@ -322,19 +322,17 @@ to generate filters to be passed to base cursor (filtersPushedDownToBaseCursor)
         // outside filter contains unnested column
         // requires check for OR and And filters, disqualify rewrite for non-unnest filters
         if (queryFilter instanceof BooleanFilter) {
-          boolean isTopLevelAndFilter = queryFilter instanceof AndFilter;
           List<Filter> preFilterList = recursiveRewriteOnUnnestFilters(
               (BooleanFilter) queryFilter,
               inputColumn,
               inputColumnCapabilites,
-              filterSplitter,
-              isTopLevelAndFilter
+              filterSplitter
           );
           // If rewite on entire query filter is successful then add entire filter to preFilter else skip and only add to post filter.
-          if (filterSplitter.getPreFilterCount() == filterSplitter.getOriginalFilterCount()) {
+          if (!preFilterList.isEmpty()) {
             if (queryFilter instanceof AndFilter) {
               filterSplitter.addPreFilter(new AndFilter(preFilterList));
-            } else if (queryFilter instanceof OrFilter) {
+            } else if (queryFilter instanceof OrFilter && filterSplitter.getPreFilterCount() == filterSplitter.getOriginalFilterCount()) {
```

Comment:
Nit. we can put a comment with an example here like if query is like
```
select * from table, UNNEST(MV_TO_ARRAY(dim3)) as u(d3) where m1=1 OR d3='a'
```
The pre filter after rewrite will be like `m1=1 OR dim3='a'` and the count of clauses in the original and pre filter matching means a successful rewrite. We can also highlight how is this for AND